src/main/java/com/example/onlinebookstore/model/Book.java

```diff
@@ -40,5 +42,9 @@ public class Book {
     private Set<Category> categories = new HashSet<>();
     @Column(name = "is_deleted")
     private boolean isDeleted;
+
+    public Book(Long id) {
+        this.id = id;
+    }
```

Comment:
redundant

Reply:
@Named(value = "bookById")
    default Book bookById(Long id) {
        return Optional.ofNullable(id)
                .map(Book::new)
                .orElse(null);
    }
    
    
    No it's usefull look at this method